### PR TITLE
basic cli arguments

### LINKED ISSUE #4: Feature: command line options
I'm willing to create a PR that allows some basic options to be configured with command line flags, such as the data dir. This would mean adding another crate as a dependency for the parsing. Is that in line with the goals of this project?

### PR DESCRIPTION
**Pull request by tversteeg**
Currently includes `[FILE]`, `--help`, `--version`, `--fullscreen` & `--data-dir`.

I've made paths global with a `OnceLock`. This is more efficient in that it doesn't need to setup and load all the paths multiple times, and was also required for persisting the paths across the whole session.

I would like to do something similar with the config ini which also seems to be loaded at multiple places, so I can expose changing things such as the window sizes. Is that ok with you?

### Examples

`cacophony -h`:

```
A minimalist and ergonomic MIDI sequencer

Usage: cacophony [OPTIONS] [FILE]

Arguments:
  [FILE]  Open the project from disk

Options:
  -d, --data-dir <DIR>  Directory where Cacophony data files reside [env: CACOPHONY_DATA_DIR=] [default: /home/thomas/r/cacophony/data]
  -f, --fullscreen      Make the window fullscreen [env: CACOPHONY_FULLSCREEN=]
  -h, --help            Print help (see more with '--help')
  -V, --version         Print version
```

`cacophony --help`:

```
A minimalist and ergonomic MIDI sequencer

Usage: cacophony [OPTIONS] [FILE]

Arguments:
  [FILE]
          Open the project from disk

Options:
  -d, --data-dir <DIR>
          Directory where Cacophony data files reside

          Uses './data' if not set

          [env: CACOPHONY_DATA_DIR=]
          [default: /home/thomas/r/cacophony/data]

  -f, --fullscreen
          Make the window fullscreen

          Uses 'fullscreen' under '[RENDER]' in 'config.ini' if not set

          Applied after displaying the splash-screen

          [env: CACOPHONY_FULLSCREEN=]

  -h, --help
          Print help (see a summary with '-h')

  -V, --version
          Print version
```

@subalterngames I've got some issues with running the actual program, so I couldn't properly test the file loading in the way I wanted. I'll open some issues for that but I would highly recommend testing my fork yourself before merging.

Closes #4.

### CODE REVIEW

File: io/src/save.rs [hunk 1]
```diff
@@ -13,7 +13,7 @@ const WRITE_ERROR: &str = "Error writing file: ";
 
 /// Serializable save data.
 #[derive(Deserialize, Serialize)]
-pub(crate) struct Save {
+pub struct Save {
```
Comment (subalterngames) [inline io/src/save.rs:16]: This should be reverted. Ideally, main.rs parses the cli args and then calls a function in io which loads the save file. I don't want to expose the save file to the other crates.

**Review by subalterngames — CHANGES_REQUESTED** [review]

**Comment by subalterngames** [comment]
@tversteeg 

> I've got some issues with running the actual program, so I couldn't properly test the file loading in the way I wanted. I'll open some issues for that but I would highly recommend testing my fork yourself before merging.

Thanks for the warning. Can you tell me specifically what doesn't work?

**Comment by tversteeg** [comment]
> Thanks for the warning. Can you tell me specifically what doesn't work?

Whenever I press any of the F keys I get a super fast flashing red text but it's too fast for me to read. I'll open a issue tomorrow when I get home. It's not related to any of my changes because I also get it on the main branch.

**Comment by subalterngames** [comment]
Thanks for the revisions. Could you fix the clippy warnings?

**Comment by tversteeg** [comment]
Fixed the clippy warnings.

**Review by subalterngames — APPROVED** [review]

**Pull request merged**